include/mega/node.h

```diff
@@ -209,53 +209,62 @@ struct MEGA_API Node : public NodeCore, FileFingerprint
     NodeCounter subnodeCounts() const;
 
     // parent
-    Node* parent;
+    Node* parent = nullptr;
 
     // children
     node_list children;
 
     // own position in parent's children
-    node_list::iterator child_it;
+    node_list::iterator child_it{};
 
     // own position in fingerprint set (only valid for file nodes)
-    Fingerprints::iterator fingerprint_it;
+    Fingerprints::iterator fingerprint_it{};
 
 #ifdef ENABLE_SYNC
     // related synced item or NULL
-    LocalNode* localnode;
+    LocalNode* localnode = nullptr;
 
     // active sync get
-    struct SyncFileGet* syncget;
+    struct SyncFileGet* syncget = nullptr;
 
     // state of removal to //bin / SyncDebris
-    syncdel_t syncdeleted;
+    syncdel_t syncdeleted = SYNCDEL_NONE;
 
     // location in the todebris node_set
-    node_set::iterator todebris_it;
+    node_set::iterator todebris_it{};
 
     // location in the tounlink node_set
     // FIXME: merge todebris / tounlink
-    node_set::iterator tounlink_it;
+    node_set::iterator tounlink_it{};
+
+    // sets whether this node can be synced to the local tree
+    void setSyncable(bool syncable);
+
+    // can this node be synced to the local tree?
+    bool isSyncable() const;
 #endif
 
     // source tag
-    int tag;
+    int tag = 0;
 
     // check if node is below this node
     bool isbelow(Node*) const;
 
     // handle of public link for the node
-    PublicLink *plink;
+    PublicLink* plink = nullptr;
 
     void setpubliclink(handle, m_time_t, m_time_t, bool);
 
     bool serialize(string*);
     static Node* unserialize(MegaClient*, string*, node_vector*);
 
+    Node();
```

Comment:
Do we need a default constructor for node?  If used, it would presumably have a null `client` pointer, which is supposed to never happen.

Reply:
That's just needed for unit testing similar to `LocalNode` and `Sync`. I will later try to refactor the tests to not require that but it may be hard.. Is this ok for now?

Reply:
Removed.